ui/ui-frontend/projects/vitamui-library/src/app/modules/components/file-selector/file-selector.component.ts

```diff
@@ -65,55 +67,64 @@ export class FileSelectorComponent {
 
   @Output() filesChanged = new EventEmitter<File[]>();
 
-  private files: File[] = [];
-  protected displayFiles: DisplayFile[] = [];
+  files: File[] = [];
+  displayFiles: DisplayFile[] = [];
+
+  constructor(
+    private translationService: TranslateService,
+    public snackBar: MatSnackBar,
+  ) {}
 
   handleFilesSelection(files: FileList | File[]) {
-    if (!this.multipleFiles && this.files.length > 0) {
+    if (!this.multipleFiles && this.files?.length > 0) {
+      this.resetInput();
+      return;
+    }
+
+    const folderNames = Array.from(files)
+      .map((file: CustomFile) => (file?.webkitRelativePath || file?.relativePath)?.split('/')[0] || file.name)
+      .filter((value, index, self) => self.indexOf(value) === index)
+      .map((folder) => folder);
+
+    if (this.isDirectoryAlreadyExists(folderNames)) {
+      this.snackBar.open(this.translationService.instant('COLLECT.UPLOAD_FILE_ALREADY_IMPORTED'), null, {
+        panelClass: 'vitamui-snack-bar',
+        duration: 10000,
+      });
+      this.resetInput();
       return;
     }
+
     // Filter to keep only the ones matching extension list (useful for drag & drop and to make sure no other type has been selected)
     const filteredFiles = Array.from(files)
       .filter((file) => !this.extensions?.length || this.extensions.some((ext) => file.name.toLowerCase().endsWith(ext.toLowerCase())))
       .slice(0, this.multipleFiles ? undefined : 1);
     this.files.push(...filteredFiles);
-    const directory = this.getDirectory(files);
-    if (this.directoryMode && directory) {
-      this.displayFiles.push(directory);
-    } else {
-      const displayFiles: DisplayFile[] = Array.from(filteredFiles).map((file: File): DisplayFile => {
-        return {
-          name: file.name,
-          size: file.size,
-          directory: false,
-        };
-      });
-      this.displayFiles.push(...displayFiles);
-    }
+    this.getDirectories(files).forEach((directory) => {
+      if (this.directoryMode && directory) {
+        this.displayFiles.push(directory);
+      } else {
+        const displayFiles: DisplayFile[] = Array.from(filteredFiles).map((file: File): DisplayFile => {
+          return {
+            name: file.name,
+            size: file.size,
+            directory: false,
+          };
+        });
+        this.displayFiles.push(...displayFiles);
+      }
+    });
```

Comment:
Le composant en mode "fichier" ne fonctionne plus.

=> il manque des tests.

Et il faut corriger :

```suggestion
    if (this.directoryMode) {
      this.displayFiles.push(...this.getDirectories(files).filter((directory) => !!directory));
    } else {
      const displayFiles: DisplayFile[] = Array.from(filteredFiles).map(
        (file: File): DisplayFile => ({
          name: file.name,
          size: file.size,
          directory: false,
        }),
      );
      this.displayFiles.push(...displayFiles);
    }
```